napari/_qt/exceptions.py

```diff
@@ -0,0 +1,107 @@
+import traceback
+from types import TracebackType
+from typing import Optional, Type
+
+from qtpy.QtCore import QObject, Signal
+from qtpy.QtWidgets import QApplication, QMainWindow, QMessageBox, QWidget
+
+from ..types import ExcInfo
+from ..utils.misc import camel_to_spaces
+
+
+class QtErrorMessageBox(QMessageBox):
+    """A messagebox that formats and displays exceptions.
+
+    Parameters
+    ----------
+    exc_info : tuple, optional
+        a tuple of ``(etype, value, tb)``. as returned by
+        :func:`sys.exc_info()`.
+    parent : QWidget, optional
+        Parent widget, by default None
+    """
+
+    def __init__(
+        self,
+        exc_info: Optional[ExcInfo] = None,
+        parent: Optional[QWidget] = None,
+    ):
+        super().__init__(parent)
+        self.setIcon(QMessageBox.Warning)
+        if exc_info:
+            self.format_exc_info(exc_info)
+
+    def format_exc_info(self, exc_info: ExcInfo):
+        """Format exception info and populate the dialog."""
+        etype, value, tb = exc_info
+        title = camel_to_spaces(etype.__name__ if etype else 'Error')
+        self.setWindowTitle(title or "Napari Error")
+        self.setText(str(value))
+
+        info = getattr(value, 'info', None)
+        if info:
+            self.setInformativeText(info + "\n")
+
+        detail = "".join(traceback.format_exception(*exc_info))
+        if detail:
+            self.setDetailedText(detail)
+            # FIXME
+            self.setStyleSheet(
+                """QTextEdit{
+                    min-width: 800px;
+                    font-size: 12px;
+                    font-weight: 400;
+                }"""
+            )
+
+
+class ExceptionHandler(QObject):
```

Comment:
I realized that NapariError is removed from the PR, is it intended? if we are to wrap sys exception handler, I would think it does make sense to introduce our wrapper error, thoughts?

Reply:
I see that as the last bullet point in https://github.com/napari/napari/issues/1395#issuecomment-657702637 ... and yes, I think that since the functionality here doesn't depend on that concept, and would be best left to a followup (as it implies a whole different discussion about what different error classes might imply for the downstream handling).